sympy/solvers/solveset.py

```diff
@@ -1116,7 +1118,16 @@ def _solveset(f, symbol, domain, _check=False):
         if isinstance(result, ConditionSet):
             # it wasn't solved or has enumerated all conditions
             # -- leave it alone
-            return result
+            if domain is S.Complexes:
+                return result
+            # Add domain
+            new_res = Intersection(result, domain)
+            # Check if the domain is really contributing
+            if new_res.is_subset(result):
```

Comment:
Isn't this always True? (Or rather is it not safe to assume that it is always True?)

Reply:
It may be that you are correct.

The thing was that just adding the domain sometimes led to some bad looking results. Typically, the domain was not really required in some cases and still remained. I do not remember an exact example, but, say, `Intersection(ConditionSet(... S.Reals), S.Reals)` or something similar. This was one of many attempts to get rid of those that at least didn't add anything "stupid".

Maybe it would be better to just add the domain unless it is `S.Complexes` and use that as input for improving `Intersection`?

Reply:
Ahh, forgot what I had already written. I'll try to find a better way of doing this.

Reply:
This was the problem:
`solveset(Abs(x) - n, x, S.Reals)` currently returns `ConditionSet(x, Contains(n, Interval(0, oo)), {-n, n})`, by Intersecting with the Reals domain we get `ConditionSet(x, Contains(_n, Interval(0, oo)), Intersection({-_n, _n}, Reals))`.

Reply:
I realize that I was trying to solve issues that I didn't fully understood. So I'll just revert this part.